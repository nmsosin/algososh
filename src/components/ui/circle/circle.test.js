import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

describe("Circle component", () => {
  it('render circle element without text', () => {
    const circleEmpty = renderer
      .create(<Circle />)
      .toJSON();

    expect(circleEmpty).toMatchSnapshot();
  });

  it('render circle element with a letter', () => {
    const circleLetter = renderer
      .create(<Circle text={'A'}/>)
      .toJSON();

    expect(circleLetter).toMatchSnapshot();
  });

  it('render circle element with head', () => {
    const circleHead = renderer
      .create(<Circle head={'head'}/>)
      .toJSON();

    expect(circleHead).toMatchSnapshot();
  });

  it('render circle element with react-element in head', () => {
    const circleHeadSubCircle = renderer
      .create(<Circle head={<Circle isSmall={true} />} />)
      .toJSON();

    expect(circleHeadSubCircle).toMatchSnapshot();
  });

  it('render circle element with tail', () => {
    const circleTail = renderer
      .create(<Circle tail={'tail'}/>)
      .toJSON();

    expect(circleTail).toMatchSnapshot();
  });

  it('render circle element with react-element in tail', () => {
    const circleTailSubCircle = renderer
      .create(<Circle tail={<Circle isSmall={true} />} />)
      .toJSON();

    expect(circleTailSubCircle).toMatchSnapshot();
  });

  it('render circle element with index', () => {
    const circleIndex = renderer
      .create(<Circle index={1}/>)
      .toJSON();

    expect(circleIndex).toMatchSnapshot();
  });

  it('render small circle element', () => {
    const circleSmall = renderer
      .create(<Circle isSmall={true}/>)
      .toJSON();

    expect(circleSmall).toMatchSnapshot();
  });

  it('render circle element with default state', () => {
    const circleStateDefault = renderer
      .create(<Circle state={ElementStates.Default}/>)
      .toJSON();

    expect(circleStateDefault).toMatchSnapshot();
  });

  it('render circle element with changing state', () => {
    const circleStateChanging = renderer
      .create(<Circle state={ElementStates.Changing}/>)
      .toJSON();

    expect(circleStateChanging).toMatchSnapshot();
  });

  it('render circle element with modified state', () => {
    const circleStateModified = renderer
      .create(<Circle state={ElementStates.Modified}/>)
      .toJSON();

    expect(circleStateModified).toMatchSnapshot();
  });
});