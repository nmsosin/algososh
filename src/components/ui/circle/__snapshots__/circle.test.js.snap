// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component render circle element with a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   changing"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circleHead"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   modified"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circleHead"
      />
      <div
        className="circle  small default"
        data-testid="circleBorder"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circleLetter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render circle element with react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="circleHead"
      />
      <div
        className="circle  small default"
        data-testid="circleBorder"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="circleLetter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="circleTail"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component render circle element with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circleTail"
  >
    tail
  </div>
</div>
`;

exports[`Circle component render circle element without text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle   default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle component render small circle element 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    className="circle  small default"
    data-testid="circleBorder"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;
